src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app.html

```diff
@@ -4,14 +4,19 @@
 >
   <!-- Local nav. -->
   <kbn-top-nav
-    ng-show="isVisible"
+    ng-show="showTopNav()"
     app-name="'dashboard'"
     config="topNavMenu"
 
-    show-search-bar="isVisible"
-    show-filter-bar="showFilterBar()"
+    show-top-nav-menu="showTopNavMenu()"
+    show-search-bar="showSearchBar()"
+    show-query-bar="showQueryBar()"
+    show-query-input="showQueryInput()"
     show-save-query="showSaveQuery"
+    show-date-picker="showDatePicker()"
+    show-filter-bar="showFilterBar()"
 
+    justify-content="justifyContent"
```

Comment:
I don't think we should be passing individual styling options here.
If absolutely necessary, we could pass `classNames`
Even better would be to let `QueryBarTopRow` decide for itself, according to which components are enabled.